app/controllers/shopping_lists_controller.rb

```diff
@@ -0,0 +1,18 @@
+class ShoppingListsController < ApplicationController
+  def index
+    @recipe = Recipe.find(params[:recipe_id])
+    @recipe_foods = @recipe.recipe_foods.includes(:food).order(:id)
+
+    @missing_ingredients = @recipe_foods.select { |ingredient| ingredient.quantity > ingredient.food.quantity }
+    @missing_ingredients = @missing_ingredients.map do |ingredient|
+      {
+        name: ingredient.food.name,
+        unit: ingredient.food.measurement_unit,
+        quantity: ingredient.quantity - ingredient.food.quantity,
+        price: ingredient.food.price * (ingredient.quantity - ingredient.food.quantity)
+      }
+    end
+
+    @total_price = @missing_ingredients.map { |ingredient| ingredient[:price] }.sum
+  end
+end
```

Comment:
- [ ] It would be nice if generate a shopping list for all the recipes not for an individual recipe :+1: I guess you didn't understand requirements properly but no worries :blush: The requirement was that you would have bunch of recipes then you would generate a single shopping list for all the missing foods in all the recipes not in one particular recipe :+1:

Reply:
Updated